File: content/LLD/Reference-Code/2. SOLID Principles/1. Single Responsibility Principle (SRP).py/2. Good Example/userRepository.py
from user import User


class UserRepository:
    def __init__(self, db, user, password):
        self.db = db
        self.user = user
        self.password = password

    def save_to_database(self, user: "User"):
        print(f"{user.name} is getting saved to database")

    def delete_from_database(self, user: "User"):
        print(f"{user.name} is getting deleted from database")

File: content/LLD/Reference-Code/2. SOLID Principles/1. Single Responsibility Principle (SRP)/2. Good Example/UserRepository.java
public class UserRepository {
    private String db;
    private String user;
    private String password;

    public UserRepository(String db, String user, String password) {
        this.db = db;
        this.user = user;
        this.password = password;
    }

    public void saveToDatabase(User user) {
        System.out.println(user.getName() + " is getting saved to database");
    }

    public void deleteFromDatabase(User user) {
        System.out.println(user.getName() + " is getting deleted from database");
    }
}
